pym/bob/input.py

```diff
@@ -2537,6 +2545,15 @@ def generatePackages(self, nameFormatter, envOverrides={}, sandboxEnabled=False)
                 except ParseError as e:
                     e.pushFrame(root.getPackageName())
                     raise e
+            for alias in self.__packageAliasList:
+                realName = self.__packageAliasList[alias]
+                try:
+                    p = walkPackagePath(result, realName)
+                    p = copy.copy(p)
+                    p.setName(alias)
```

Comment:
This is where the internal structure falls apart. The package name will not match the recipe name. Also the Steps back-link to their Package and this is broken too. All in all I think this is the wrong approach.